### PR TITLE
If the content length is already set by a previous handler, then unset it

### LINKED ISSUE #17: net::ERR_CONTENT_LENGTH_MISMATCH
Hey, if I take the example in the readme and try to use it with a static directory server I get a `net::ERR_CONTENT_LENGTH_MISMATCH` error in the browser. Here's the example with the test code I've added:

```go
package main

import (
	"io"
	"net/http"
	"os"

	"github.com/nytimes/gziphandler"
)

func main() {
	withoutGz := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "text/plain")
		io.WriteString(w, "Hello, World")
	})

	withGz := gziphandler.GzipHandler(withoutGz)

	dirWithoutGz := http.FileServer(http.Dir("./"))
	dirWithGz := gziphandler.GzipHandler(dirWithoutGz)

	http.Handle("/", withGz)
	http.Handle("/test.jpg", dirWithGz)
	http.ListenAndServe(":"+os.Getenv("PORT"), nil)
}
```

If you use the `dirWithoutGz` handler then the image returns just fine, but the `dirWithGz` handler errors. Am I misunderstanding something?

Here's some debug information if this helps:

```sh
$GOPATH/src/github.com/nytimes/gziphandler master
❯ git rev-parse HEAD
44668d75e46f05932cf7c1c7a375d0765b324a0b
```

```sh
❯ go version
go version go1.7.1 darwin/amd64
```

### PR DESCRIPTION
**Pull request by fastest963**
Fixes #17 and #12

### CODE REVIEW

File: gzip_test.go [hunk 1]
```diff
@@ -167,6 +169,50 @@ func TestGzipHandlerNoBody(t *testing.T) {
...
+		t.Fatalf("Unexpected error making http request: %v", err)
+	}
+	defer res.Body.Close()
+
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Fatalf("Unexpected error reading response body: %v", err)
+	}
+
+	len, err := strconv.Atoi(res.Header.Get("Content-Length"))
```
Comment (jprobinson) [inline gzip_test.go]: Mind renaming this variable so we don't confuse it with the `len()` builtin func?

File: gzip_test.go [hunk 2]
```diff
@@ -167,6 +169,50 @@ func TestGzipHandlerNoBody(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("Unexpected error parsing Content-Length: %v", err)
+	}
+	assert.Len(t, body, len)
```
Comment (jprobinson) [inline gzip_test.go]: We're asserting the length returned matches the length of the body, but not asserting that the body returned actually _is_ gzipped. I guess it's not totally needed (I'm thinking I'm paranoid), but would you mind tossing in an assert that verifies that `body` and the original string returned by the handler aren't the same?

**Comment by fastest963** [comment]
@jprobinson thanks for the quick review. Just pushed up the changes you requested. I decided to cehck the `Content-Encoding` and added a `NotEqual`

**Comment by jprobinson** [comment]
🍕 :boom: thanks again!

**Comment by fastest963** [comment]
Thanks for the quick merge @jprobinson! This also should fix #12 by the way.

**Pull request merged**